Doc/library/asyncio-stream.rst

```diff
@@ -160,11 +184,87 @@ and work with streams:
 
    .. deprecated-removed:: 3.8 3.10
 
-      `start_unix_server()` is deprecated in favor of `UnixStreamServer()`.
+      `start_unix_server()` is deprecated in favor of :class:`UnixStreamServer`.
 
 
 ---------
 
+StreamServer
+============
+
+.. class:: StreamServer(client_connected_cb, /, host=None, port=None, *, \
+                        limit=2**16, family=socket.AF_UNSPEC, \
+                        flags=socket.AI_PASSIVE, sock=None, backlog=100, \
+                        ssl=None, reuse_address=None, reuse_port=None, \
+                        ssl_handshake_timeout=None, shutdown_timeout=60)
+
+   .. coroutinefunction:: start_serving
+
+      Binds to the given host and port to start the server. This method is
+      automatically called during ``__enter__`` when :class:`Stream` is
+      used as a context manager.
+
+   .. coroutinefunction:: close
+
+      Closes the connection. This method is automatically called during
+      ``__exit__`` when :class:`StreamServer` is used as a context manager.
+
+   .. method:: is_serving
+
+      Returns ``True`` if the server is bound and currently serving.
+
+   .. attribute:: sockets
+
+      Returns a tuple of sockets the server is bound to.
+
+   .. versionadded:: 3.8
+
+
+UnixStreamServer
+================
+
+.. class:: UnixStreamServer(client_connected_cb, /, path=None, *, \
+                            limit=2**16, sock=None, backlog=100, \
+                            ssl=None, ssl_handshake_timeout=None, shutdown_timeout=60)
+
+   .. coroutinefunction:: start_serving
+
+      Binds to the given host and port to start the server. This method is
+      automatically called during ``__enter__`` when :class:`Stream` is
+      used as a context manager.
+
+   .. coroutinefunction:: close
+
+      Closes the connection. This method is automatically called during
+      ``__exit__`` when :class:`UnixStreamServer` is used as a context manager.
+
+   .. method:: is_serving
+
+      Returns ``True`` if the server is bound and currently serving.
+
+   .. attribute:: sockets
+
+      Returns a tuple of sockets the server is bound to.
+
+   .. availability:: Unix.
+
+   .. versionadded:: 3.8
+
+Stream
+============
+
+.. class:: Stream
+
+   Represents a Stream object that provides APIs to read and write data
+   to the IO stream . It includes the API provided by :class:`StreamReader`
+   and :class:`StreamWriter`.
+
+   It is not recommended to instantiate *Stream* objects
```

Comment:
`not recommended` sounds like "you shouldn't but you can".
But the stream *must* be instantiated by asyncio only.

I think the text should be
```
   Do not instantiate *Stream* objects
   directly; use API like :func:`connect` and :class:`StreamServer`
   instead.
```
I don't think we should enumerate all `connect_unix()` / `UnixStreamServer` / `connect_read_pipe()` / `connect_write_pipe()` here

Reply:
Thanks, I have added your suggestion.